src/Vimeo/ABLincoln/Namespaces/SimpleNamespace.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Vimeo\ABLincoln\Namespaces;
+
+use \Psr\Log\LoggerInterface;
+
+/**
+ * Simple namespace base class that handles user assignment when dealing with
+ * multiple concurrent experiments. Exposure logs according to PSR-3 logging
+ * specifications. User experiments extending this class should pass in their
+ * own compatible logger instance.
+ */
+abstract class SimpleNamespace
```

Comment:
I think you're right - when I first wrote this I was operating under a multiple-inheritance assumption but I've noticed that traits really are more useful for adding smaller bits of related functionality (e.g. PSR logging or file logging). Vimeo single-inherits from a data store object in order to read experiment info from the database, so it eased our integration there to treat everything as a trait instead of a class (essentially allowing multiple inheritance from the DB object and the planout experiment classes as well). But this increasingly seems to me like a special case and probably not one we should build the open source version around.

After a bit more thought I'm thinking of consolidating into the following structure (to better mirror the Python version and use traits a bit more systematically):

```
Experiments/
  |-- Logging/
    |-- PSRLoggerTrait
    |-- FileLoggerTrait (uses PSRLoggerTrait + logger to log to file)
  |-- AbstractExperiment (mirror of the Experiment class in Python)
  |-- DefaultExperiment (unchanged)
  |-- SimpleExperiment (uses FileLoggerTrait)
Namespaces/
  |-- AbstractNamespace (provides higher level methods + implements addExperiment etc. in terms of these methods)
  |-- SimpleNamespace (implements higher level methods using dictionaries)
```

As you can see I yet to do anything with the interpreter but ideally that would be a nice next step. I'm also interested to hear what @tsujeeth and @GregBowyer think!